examples/demo/src/controllers/cache.rs

```diff
@@ -11,14 +11,33 @@ async fn get_cache(State(ctx): State<AppContext>) -> Result<Response> {
         value: ctx.cache.get("value").await.unwrap(),
     })
 }
+
 async fn insert(State(ctx): State<AppContext>) -> Result<Response> {
     ctx.cache.insert("value", "loco cache value").await.unwrap();
     format::empty()
 }
 
+async fn get_or_insert(State(ctx): State<AppContext>) -> Result<Response> {
+    let res = ctx
+        .cache
+        .get_or_insert("user", async {
```

Comment:
something still seems off to me w/errors.

i believe, async block should just throw the regular Result from Loco (i.e. `CacheError` does not belong in that async block)

the block runs like any other "linear" code flow, only that in our case, the result of the block is cached. So the block can return a result or fail, like any other code flow. The failure should be for any *original* reason the block itself has failed.

a good way to look at it:

if you remove the cache completely, the code block should mostly stay the same

Reply:
got you, changed for supporting it